From ec6facd380f591f0c6d77b65eea5eff48ef01229 Mon Sep 17 00:00:00 2001
From: "Hebbar, Gururaja" <gururaja.hebbar@ti.com>
Date: Tue, 5 Jun 2012 09:31:33 +0530
Subject: [PATCH 1019/1092] ARM: OMAP2+: am335x: Manually configure CLKOUT1
 pin

CLKOUT1 pin will allow user to extract raw oscillator clock
(Master_OSC). In order to achieve this SYSBOOT[5] must be set. When
SYSBOOT[5] is set to 1, the SOC will automatically setup the related
pin-mux to mode 3.

However, on some boards (like EVM-SK), SYSBOOT[5] isn't properly
latched. Hence setting clkout1 pin-mux manually will solve the
problem.

Signed-off-by: Hebbar, Gururaja <gururaja.hebbar@ti.com>
---
 arch/arm/mach-omap2/board-am335xevm.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/arch/arm/mach-omap2/board-am335xevm.c b/arch/arm/mach-omap2/board-am335xevm.c
index 09d4517..43d5ac9 100644
--- a/arch/arm/mach-omap2/board-am335xevm.c
+++ b/arch/arm/mach-omap2/board-am335xevm.c
@@ -184,6 +184,16 @@ static struct omap2_hsmmc_info am335x_mmc[] __initdata = {
 
 #ifdef CONFIG_OMAP_MUX
 static struct omap_board_mux board_mux[] __initdata = {
+	/*
+	 * Setting SYSBOOT[5] should set xdma_event_intr0 pin to mode 3 thereby
+	 * allowing clkout1 to be available on xdma_event_intr0.
+	 * However, on some boards (like EVM-SK), SYSBOOT[5] isn't properly
+	 * latched.
+	 * To be extra cautious, setup the pin-mux manually.
+	 * If any modules/usecase requries it in different mode, then subsequent
+	 * module init call will change the mux accordingly.
+	 */
+	AM33XX_MUX(XDMA_EVENT_INTR0, OMAP_MUX_MODE3 | AM33XX_PIN_OUTPUT),
 	AM33XX_MUX(I2C0_SDA, OMAP_MUX_MODE0 | AM33XX_SLEWCTRL_SLOW |
 			AM33XX_INPUT_EN | AM33XX_PIN_OUTPUT),
 	AM33XX_MUX(I2C0_SCL, OMAP_MUX_MODE0 | AM33XX_SLEWCTRL_SLOW |
-- 
1.7.11.2

